### PR TITLE
[Merged by Bors] - chore: backport fixes to deprecations in dot notation

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean [hunk 1]
```diff
@@ -94,6 +94,7 @@ instance Measure.instFunLike [MeasurableSpace α] : FunLike (Measure α) (Set α
 
 #noalign measure_theory.measure.has_coe_to_fun
 
+set_option linter.deprecated false in -- Not immediately obvious how to use `measure_empty` here.
```
Comment (Ruben-VandeVelde) [inline Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: @urkud

**Comment by YaelDillies** [comment]
maintainer merge

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean (hunk 1 above)
Comment (jcommelin) [inline Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: Maybe something like:
```suggestion
#adaptation_note /-- (4.9.0) Not immediately obvious how to use `measure_empty` here. -/
set_option linter.deprecated false in
```
Please rephrase!

File: Mathlib/MeasureTheory/OuterMeasure/Caratheodory.lean [hunk 2]
```diff
@@ -111,6 +111,7 @@ theorem isCaratheodory_sum {s : ℕ → Set α} (h : ∀ i, IsCaratheodory m (s
     simpa using fun (h₁ : a ∈ s n) i (hi : i < n) h₂ => (hd (ne_of_gt hi)).le_bot ⟨h₁, h₂⟩
 #align measure_theory.outer_measure.is_caratheodory_sum MeasureTheory.OuterMeasure.isCaratheodory_sum
 
+set_option linter.deprecated false in -- not immediately obvious how to replace `iUnion` here.
```
Comment (jcommelin) [inline Mathlib/MeasureTheory/OuterMeasure/Caratheodory.lean:114]: Idem
```suggestion
#adaptation_note  /-- (4.9.0) not immediately obvious how to replace `iUnion` here. -/
set_option linter.deprecated false in
```

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: I disagree with using `#adaptation_note` here. This has nothing to do with v4.9.0: it's always been deprecated, the only new thing is that v4.9.0 will tell us about this.

Comment (semorrison) [reply Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: I think `set_option linter.deprecated false` is exactly the annotation we need to keep track of these defects!

**Comment by semorrison** [comment]
I'm going to merge, despite not following the last request from the delegator, because this is blocking updating bump/v4.9.0. I'll pick up this thread on zulip.

bors p=10

bors merge

**Pull request closed without merging**